### DISCUSSION TITLE
Errors with blade components included within livewire components

### DISCUSSION THREAD

**Discussion by OccTherapist** [livewire/livewire#5164] [1. Help]
Hi,

I updated to Laravel 9 when experiencing errors within my livewire components. Everytime I include a blade component via `<x-component>` syntax.

The error which says 

> Call to a member function getIterator() on array

But the error disappears when I include my components manually into the livewire component.

I spent a whole bunch of time digging and googling into this issue, thinking it was my mistake. But in the end I believe this must be a bug.

</br>

### The postsearch Livewire component

```<?php

namespace App\Http\Livewire;

use Livewire\Component;
use App\Models\Blog\Post;

class Postsearch extends Component
{
    public $attributes = [];
    public $query;
    public $posts = [];

    public function search(): void
    {
        if (! $this->query || ! $this->attributes) {
            $this->posts = [];
            return;
        }

        $this->posts = Post::query()
            ->select('id', 'title', 'slug', 'created_at', updated_at')
            ->with('tags')
            ->whereLike($this->attributes, $this->query)
            ->get();
    }

    public function render()
    {
        return view('livewire.postsearch', ['searchResults' => $this->posts]);
    }
}


```
### included like this

```
<x-collapsable.wrapper>
        <x-collapsable.item title="Posts Suche">
            <livewire:postsearch />
        </x-collapsable.item>
</x-collapsable.wrapper>
```

### the component view

```
<div>
    <form class="form" wire:submit.prevent="search" method="post">
        
        // here is where the first error happens (first blade component)
        <x-form.textfield
            name="query"
            labelText="Stichwort"
            labelAlwaysFloating="true"
            wire:model.defer="query"
        />

        <p>Wo soll das Stichwort enthalten sein?</p>
        <fieldset>
            <label class="label__checkbox">
                <span>Titel</span>
                <input
                    type="checkbox"
                    name="attributes"
                    id="title"
                    value="title"
                    checked="checked"
                    wire:model="attributes"
                >
            </label>

           [...]  // more of this

        </fieldset>

        <div class="button-wrapper">
            <button class="btn btn-primary" wire:click="search">Suchen</button>
        </div>

        <x-form.script />
    </form>

    <div class="loading-wrapper">
        <div wire:loading.delay class="loading">
            <x-icons.loading />
            <p>Lade Ergebnisse&hellip;</p>
        </div>
    </div>

    @if ($searchResults)
          [...] // display results
    @endif
</div>
```

### Screenshot of the error
<a href="https://s20.directupload.net/images/220807/3s599jxn.png" target="_blank"><img src="https://s20.directupload.net/images/220807/3s599jxn.png" title="Kostenlos Bilder und Fotos hochladen"></a>

When I replace all `<x-blade-components>` with the HTML it works fine. 

Kind regards

**Comment by joshhanley** [comment]
@OccTherapist hard to say without seeing some code 🙂 Could you please share your code which reproduces the issue with as little code as possible so we can investigate it? Thanks!

**Reply by OccTherapist** [reply]
@joshhanley thank you for the fast answer. I added the code and more information above.

**Reply by joshhanley** [reply]
@OccTherapist all looks good to me (for the code that I can see)!

Can you create a new Livewire component and Blade component and see if you can get them to work together on a new test route?

If that doesn't work, can you share the new component class and view as well as the blade component, so I can test it locally?

Thanks!

**Reply by OccTherapist** [reply]
@joshhanley it works. Unfortunately... xD

**Comment by OccTherapist** [comment]
@joshhanley created new test components and route. It worked as expected...

### The Route
`Route::view('/test', 'test');`

### The View
```
<x-layouts.main title="TEST">
    <h1>TEST</h1>

    <livewire:test />
</x-layouts.main>
```

### The Livewire Component
```
<?php

namespace App\Http\Livewire;

use Livewire\Component;

class Test extends Component
{
    public function render()
    {
        return view('livewire.test');
    }
}
```

```
<div>
    {{-- Be like water. --}}
    <x-test/>
</div>
```

### The Blade Component
```
<div>
    <!-- The best way to take care of the future is to take care of the present moment. - Thich Nhat Hanh -->
    <span>TEST 8 August 2022</span>
</div>

```

No error.

**Reply by joshhanley** [reply]
@OccTherapist ok well that's good in a way, it means it's not a bug with Livewire 😁

So something in your code must be causing the issue. I would go though removing each Blade component from your Livewire view until you can work out which Blade component is breaking, and then see if you can work out why or post up the code for that Blade component and I can have a look to see if I can spot the issue 🙂

**Reply by OccTherapist** [reply]
@joshhanley damn, I tried this already. Will try it again tomorow, Maybe I missed something.

**Reply by joshhanley** [reply]
@OccTherapist bugger!

Not sure how you did it last time, but I would just go through each blade component include in the Livewire view, one by one, comment out the line for the blade component include, refresh browser and see if error happens, and repeat. Hopefully that will narrow it down to which one it is!

**Reply by OccTherapist** [reply]
@joshhanley actually the irritating thing is on this specific livewire component I get this error on every included blade-x component even on svg icons I use without any variable in it. 

### This thing errors
` <x-icons.loading />`

### The included HTML is just a vector
```
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="150px" height="150px" viewBox="0 0 100 100" preserveAspectRatio="xMidYMid">
<g transform="rotate(0 50 50)">
  <rect x="41.5" y="31.5" rx="8.5" ry="1.5" width="17" height="3" fill="#ffb6bb">
    <animate attributeName="opacity" values="1;0" keyTimes="0;1" dur="1.075268817204301s" begin="-0.8960573476702509s" repeatCount="indefinite"></animate>
  </rect>
</g><g transform="rotate(60 50 50)">
  <rect x="41.5" y="31.5" rx="8.5" ry="1.5" width="17" height="3" fill="#ffe691">
    <animate attributeName="opacity" values="1;0" keyTimes="0;1" dur="1.075268817204301s" begin="-0.7168458781362007s" repeatCount="indefinite"></animate>
  </rect>
</g><g transform="rotate(120 50 50)">
  <rect x="41.5" y="31.5" rx="8.5" ry="1.5" width="17" height="3" fill="#95d5ee">
    <animate attributeName="opacity" values="1;0" keyTimes="0;1" dur="1.075268817204301s" begin="-0.5376344086021505s" repeatCount="indefinite"></animate>
  </rect>
</g><g transform="rotate(180 50 50)">
  <rect x="41.5" y="31.5" rx="8.5" ry="1.5" width="17" height="3" fill="#585872">
    <animate attributeName="opacity" values="1;0" keyTimes="0;1" dur="1.075268817204301s" begin="-0.35842293906810035s" repeatCount="indefinite"></animate>
  </rect>
</g><g transform="rotate(240 50 50)">
  <rect x="41.5" y="31.5" rx="8.5" ry="1.5" width="17" height="3" fill="#9e4446">
    <animate attributeName="opacity" values="1;0" keyTimes="0;1" dur="1.075268817204301s" begin="-0.17921146953405018s" repeatCount="indefinite"></animate>
  </rect>
</g><g transform="rotate(300 50 50)">
  <rect x="41.5" y="31.5" rx="8.5" ry="1.5" width="17" height="3" fill="#ffb6bb">
    <animate attributeName="opacity" values="1;0" keyTimes="0;1" dur="1.075268817204301s" begin="0s" repeatCount="indefinite"></animate>
  </rect>
</g>
</svg>
```

**Reply by joshhanley** [reply]
@OccTherapist hmm how strange! Can you share the code fro your component's complete view and class (plus anything else you may think I'll need) and I'll copy it down and see if I get the same error?

**Reply by OccTherapist** [reply]
Hey, sry I've been very busy lately. This is still an issue and I will come back to it when working next on that specific project. Have to finish other work first, though. Kind regards Igor

**Reply by OccTherapist** [reply]
Hey @joshhanley, sorry for the late answer. I'm back at this codebase againg and the issue persists so far. 
 
 I have had the assumption, that livewire maybe has problems with included blade-x-components which don*t have a class representation in `App\Views`. Because all included components in this view hadn't got any class. But then I tested it with an newly created component and a representing class and still got the same error.

**Reply by OccTherapist** [reply]
The `@include` directive with an array of attributes still works. It somehow just has a problem with `<x-blade/>` syntax.

**Comment by CodyPChristian** [comment]
@OccTherapist @joshhanley I fought with this for awhile myself and found the root cause. In both my case and Occ's we used `$attributes` in our livewire comp which at some point you can no longer do. I think this may have been a L9 change as I'm starting to notice the problem only with my L9 apps.

**Reply by joshhanley** [reply]
@CodyPChristian ah interesting! Any chance you could share some code that replicates the issue so we can have a look? Thanks!
